engine-nextgen/src/main/kotlin/graphql/nadel/validation/NadelTypeValidation.kt

```diff
@@ -198,6 +209,19 @@ internal class NadelTypeValidation(
             } to errors
     }
 
+    private fun getSyntheticUnions(service: Service): Set<GraphQLUnionType> {
+        return service.definitionRegistry
+            .definitions
+            .asSequence()
+            .filterIsInstance<ObjectTypeDefinition>()
+            .flatMap { it.fieldDefinitions }
+            .filter { it.getDirectives(HYDRATED_DIRECTIVE_DEFINITION.name).size > 1 }
+            .map { it.type.unwrapAll() }
+            .map { overallSchema.getType(it.name) }
+            .filterIsInstance<GraphQLUnionType>()
+            .toSet()
+    }
+
```

Comment:
naming is hard - but are they "synthetic" (eg made up -  no real) - I mean they are multi service sure

I think `getPolymophicHydrationUnions` might be a better name